mavros/src/plugins/imu.cpp

```diff
@@ -238,10 +238,18 @@ class IMUPlugin : public plugin::PluginBase {
 		tf::vectorEigenToMsg(gyro_flu, imu_msg->angular_velocity);
 		tf::vectorEigenToMsg(accel_flu, imu_msg->linear_acceleration);
 
-		// Save readings
-		linear_accel_vec_flu = accel_flu;
-		linear_accel_vec_frd = accel_frd;
-		received_linear_accel = true;
+		if (!m_uas->is_ardupilotmega()) {
```

Comment:
`~imu/data` should be consistent across different FCUs, so it's safer not to publish anything, than unscaled values.

Reply:
Yes, ~imu/data_raw will publish unscaled values, and ~imu/data will keep default value {0};
m_uas->is_ardupilotmega() will become True actually latter, but after the heartbeat publish the correct pilot type.